mlptrain/sampling/reaction_coord.py

```diff
@@ -122,3 +128,82 @@ def _grad(self, atoms: ase.atoms.Atoms):
     def n_pairs(self) -> int:
         """Number of pairs of atoms in this set"""
         return len(self.atom_pair_list)
+
+
+class DifferenceDistance(ReactionCoordinate):
+    """Difference in distance between two pairs of specified atoms"""
+
+    def __init__(self, *args):
+        """
+        Difference of a pair of distances e.g.
+
+        # Δr = r_1 - r_2, where r_i is the distance between atoms j and k
+        dists = DifferenceDistance((0, 1), (0, 2))
+
+        Must comprise two pairs of atoms
+
+        -----------------------------------------------------------------------
+        Arguments:
+            args: Pairs of atom indices
+        """
+
+        self.atom_pair_list = []
+
+        if len(args) != 2:
+            raise ValueError('Difference Distance must comprise exactly two '
+                             'pairs of atoms')
+
+        for arg in args:
+            if len(arg) != 2:
+                raise ValueError('Average distance must be initialised from '
+                                 'a 2-tuple of atom indices')
+
+            self.atom_pair_list.append(tuple(arg))
+
+    def _call(self, atoms: ase.atoms.Atoms):
+        """Difference in distance between two atom pairs"""
+        dists = [atoms.get_distance(i, j, mic=True)
+                 for (i, j) in self.atom_pair_list]
+
+        return dists[0] - dists[1]
+
+    def _grad(self, atoms: ase.atoms.Atoms):
+        """Gradient of the difference in distance between two atom pairs. Each
+        component of the gradient is calculated using ∇B_i,m:
+
+        ∇B_i,m = (r_i,m - r_i,m') / ||r_mm'||
+
+        An additional term is subtracted if there are only 3 atoms in the 2
+        pairs. e.g. - (r_i,m - r_i,n) / ||r_mn||
+
+        ∇B_i,m:  Gradient of bias for atom m along component i for pair m, m'
+        r_i,m:   i (= x, y or z) position of atom in pair m, m'
+        ||r_m||: Euclidean distance between atoms in pair m, m'
+        """
+
```

Comment:
Not sure how important performance is but this can probably turn this into a single numpy call

Reply:
What do you mean here sorry?

Reply:
see https://github.com/duartegroup/mlp-train/pull/19#discussion_r942718470 (github is stilly with suggestions)